src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx

```diff
@@ -242,36 +243,39 @@ export const IndexPatternTable = ({ canSave, history }: Props) => {
   }
 
   return (
-    <EuiPageContent data-test-subj="indexPatternTable" role="region" aria-label={ariaRegion}>
-      <EuiFlexGroup justifyContent="spaceBetween">
-        <EuiFlexItem grow={false}>
-          <EuiTitle>
-            <h2>{title}</h2>
-          </EuiTitle>
-          <EuiSpacer size="s" />
-          <EuiText>
-            <p>
-              <FormattedMessage
-                id="indexPatternManagement.indexPatternTable.indexPatternExplanation"
-                defaultMessage="Create and manage the index patterns that help you retrieve your data from OpenSearch."
-              />
-            </p>
-          </EuiText>
-        </EuiFlexItem>
-        <EuiFlexItem grow={false}>{createButton}</EuiFlexItem>
-      </EuiFlexGroup>
-      <EuiSpacer />
-      <EuiInMemoryTable
-        allowNeutralSort={false}
-        itemId="id"
-        isSelectable={false}
-        items={indexPatterns}
-        columns={columns}
-        pagination={pagination}
-        sorting={sorting}
-        search={search}
-      />
-    </EuiPageContent>
+    <>
+      {dataSourceEnabled ? <ExperimentalCallout /> : null}
```

Comment:
Looks like we may not need this logic since the list should not be rendered in the first place if `dataSourceEnabled` is false. See line 221

Reply:
If you look at line 220, those 2 empty state components are returned only when there are NO index patterns in the table. If there are any index patterns created earlier or using default, then the table component would be displayed.
So my condition is that, if table shows up & data source is enabled, it should display experimental call out.
When index patterns length is 0, then create index pattern prompt will show up and when user clicks on create button, on create page experimental call out will be displayed.

Reply:
Oh okay, if thats the case, a unit test here would be more useful than the snapshot test you have for the `ExperimentalCallout` component. You want to be testing logic and not the text itself since that may change and that will be intentional.

Will not block on it though.

Reply:
Agree. We will have another PR for unit tests coverage since we were working on several same files. And once these two PRs has been merged, I will start to update unit test accordingly.